webapp/channels/src/components/widgets/settings/text_setting.tsx

```diff
@@ -1,108 +1,91 @@
-        inputClassName: '',
-        type: 'input',
-        maxLength: -1, // A negative number allows for values of any length
-        resizable: true,
-    };
-
-    private handleChange: HandleChangeTypes = (e) => {
-        if (this.props.type === 'number') {
-            this.props.onChange(this.props.id, parseInt(e.target.value, 10));
+    function handleChange(event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
+        if (props.type === 'number') {
+            props.onChange(props.id, parseInt(event.target.value, 10));
         } else {
-            this.props.onChange(this.props.id, e.target.value);
-        }
-    };
-
-    public render(): JSX.Element {
-        const {resizable} = this.props;
-        let {type} = this.props;
-        let input = null;
-
-        if (type === 'textarea') {
-            let style = {};
-            if (!resizable) {
-                style = Object.assign({}, {resize: 'none'});
-            }
-
-            input = (
-                <textarea
-                    autoFocus={this.props.autoFocus}
-                    data-testid={this.props.id + 'input'}
-                    id={this.props.id}
-                    dir='auto'
-                    style={style}
-                    className='form-control'
-                    rows={5}
-                    placeholder={this.props.placeholder}
-                    value={this.props.value}
-                    maxLength={this.props.maxLength}
-                    onChange={this.handleChange}
-                    disabled={this.props.disabled}
-                />
-            );
-        } else {
-            type = ['input', 'email', 'tel', 'number', 'url', 'password'].includes(type) ? type : 'input';
-
-            input = (
-                <input
-                    autoFocus={this.props.autoFocus}
-                    data-testid={this.props.id + type}
-                    id={this.props.id}
-                    className='form-control'
-                    type={type}
-                    placeholder={this.props.placeholder}
-                    value={this.props.value}
-                    maxLength={this.props.maxLength}
-                    onChange={this.handleChange}
-                    disabled={this.props.disabled}
-                />
-            );
+            props.onChange(props.id, event.target.value);
         }
+    }
 
-        return (
-            <Setting
-                label={this.props.label}
-                labelClassName={this.props.labelClassName}
-                inputClassName={this.props.inputClassName}
-                helpText={this.props.helpText}
-                inputId={this.props.id}
-                footer={this.props.footer}
-            >
-                {input}
-            </Setting>
+    let input = null;
+    if (type === 'textarea') {
+        input = (
+            <textarea
+                autoFocus={props.autoFocus}
+                data-testid={`${props.id}--${type}`}
```

Comment:
Do any E2E tests need to be updated for this change?

Reply:
reverted back to old id